### PR TITLE
tweak fontsizes

### PR DESCRIPTION
**Pull request by Jay-o-Way**
## Description

Just small changes

## Motivation and Context

- Do not to use font-size 14 on icons, they are designed with 16px in mind.
  - only change properties when needed, but at the very minimum use the "rule-of-4"
- Quote from the app: "The minimum values should be 12px Regular, 14px Semibold"

## Screenshots (if appropriate):

WinUI3Gallery icons

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by karenbtlai** [comment]
Hey Jay-o-Way! Thanks for your contribution. Do you mind adding more detailed before and after changes photos in your PR description?

**Comment by Jay-o-Way** [comment]
Hi @karenbtlai I would love to. But I have issues actually running/debugging both the projects.
WinUIGallery.DesktopWap gives me:
image

WinUIGallery.DesktopWap.Package gives me:
image

**Comment by Jay-o-Way** [comment]
@karenbtlai done. Visually, it's absolutely minute. More a matter of principle and design/code habits.

File: WinUIGallery/Controls/PageHeader.xaml [hunk 1]
```diff
@@ -48,7 +48,7 @@
                     Visibility="{x:Bind Item.Docs, Converter={StaticResource collectionConverter}}">
                     <DropDownButton.Content>
                         <StackPanel Orientation="Horizontal" Spacing="8">
-                            <FontIcon FontSize="14" Glyph="&#xE8A5;" />
+                            <FontIcon Glyph="&#xE8A5;" />
```
Comment (niels9001) [inline WinUIGallery/Controls/PageHeader.xaml:51]:
```suggestion
                            <FontIcon FontSize="16" Glyph="&#xE8A5;" />
```

File: WinUIGallery/Controls/PageHeader.xaml [hunk 2]
```diff
@@ -116,7 +116,7 @@
                     ToolTipService.ToolTip="Toggle theme"
                     Visibility="{x:Bind ThemeButtonVisibility, Mode=OneWay}">
                     <Button.Content>
-                        <FontIcon FontSize="14" Glyph="&#xE793;" />
+                        <FontIcon Glyph="&#xE793;" />
```
Comment (niels9001) [inline WinUIGallery/Controls/PageHeader.xaml:119]:
```suggestion
                        <FontIcon FontSize="16" Glyph="&#xE793;" />
```

File: WinUIGallery/Controls/PageHeader.xaml [hunk 3]
```diff
@@ -155,7 +154,7 @@
                     Click="OnFeedBackButtonClick"
                     ToolTipService.ToolTip="Send feedback">
                     <Button.Content>
-                        <FontIcon FontSize="14" Glyph="&#xED15;" />
+                        <FontIcon Glyph="&#xED15;" />
```
Comment (niels9001) [inline WinUIGallery/Controls/PageHeader.xaml:157]:
```suggestion
                        <FontIcon FontSize="16" Glyph="&#xED15;" />
```

**Review by niels9001 — CHANGES_REQUESTED** [review]

**Comment by niels9001** [comment]
@jay-o-way @karenbtlai 

Synced with the design team, and by default we should adhere to using 16/24/32/48 as @Jay-o-Way pointed out.

However, FontIcon/SymbolIcon uses 20 by default (I always was under the impression it would default to 16, like normal text..). So we still have to manually set these to "16".

Chart for comparison (as @Jay-o-Way pointed out, 14 causes blurriness on higher resolutions screens):
image

This is also inline with icons in e.g. the Start Menu or systemtray:
  
image

**Comment by Jay-o-Way** [comment]
> FontIcon/SymbolIcon uses 20 by default

Could that be, because of a style somewhere, perhaps?

**Comment by niels9001** [comment]
> > FontIcon/SymbolIcon uses 20 by default
> 
> Could that be, because of a style somewhere, perhaps?

Yeah, I'm guessing this might be from the pre-WinUI era where in general icons and fonts were bigger.

File: WinUIGallery/ControlPages/ListViewPage.xaml [hunk 4]
```diff
@@ -429,7 +429,7 @@ sent or received, and those values are bound in the DataTemplate.--&gt;
             &lt;StackPanel Orientation="Horizontal"&gt;
                 &lt;TextBlock Text="{x:Bind Views}" HorizontalAlignment="Left" Style="{ThemeResource CaptionTextBlockStyle}" Margin="0,0,0,0"/&gt;
                 &lt;TextBlock Text=" Views " HorizontalAlignment="Left" Style="{ThemeResource CaptionTextBlockStyle}"/&gt;
-                &lt;TextBlock Text=" &#x22C5; " FontFamily="Segoe UI" FontSize="12" FontWeight="Bold" 
+                &lt;TextBlock Text=" &#x22C5; " FontFamily="Segoe UI" Style="{ThemeResource CaptionTextBlockStyle}"
```
[the remark this answers is not in the document]
Comment (Jay-o-Way) [reply WinUIGallery/ControlPages/ListViewPage.xaml]: Done

**Comment by Jay-o-Way** [comment]
So, pr looks good now? :)

**Comment by niels9001** [comment]
@jay-o-way

Icons seem to be vertically off center now..
image

File: WinUIGallery/Controls/PageHeader.xaml [hunk 5]
```diff
@@ -48,7 +48,7 @@
                     Visibility="{x:Bind Item.Docs, Converter={StaticResource collectionConverter}}">
                     <DropDownButton.Content>
                         <StackPanel Orientation="Horizontal" Spacing="8">
-                            <FontIcon FontSize="14" Glyph="&#xE8A5;" />
+                            <FontIcon FontSize="16" Glyph="&#xE8A5;" />
                             <TextBlock Text="Documentation" />
```
Comment (niels9001) [inline WinUIGallery/Controls/PageHeader.xaml:52]: You might want to add a bottom margin (`Margin="0,0,0,2") to the `TextBlock` so it's centered correctly.

File: WinUIGallery/Controls/PageHeader.xaml [hunk 6]
```diff
@@ -129,7 +129,6 @@
                     <Button.Content>
                         <FontIcon
                             x:Name="CopyLinkButtonIcon"
```
Comment (niels9001) [inline WinUIGallery/Controls/PageHeader.xaml:131]: This would need `FontSize="16"` as well?

**Review by niels9001 — CHANGES_REQUESTED** [review]

File: WinUIGallery/Controls/PageHeader.xaml (hunk 5 above)
Comment (Jay-o-Way) [reply WinUIGallery/Controls/PageHeader.xaml:52]: > looks like it's the link icon

@niels9001 You've got sharp eyes!
I took the image from https://learn.microsoft.com/nl-nl/windows/apps/design/style/segoe-fluent-icons-font#:~:text=Stop-,e71b,-Link
and see that this is by design. If Microsoft designed it like that, I'm nog touching it :)
image

File: WinUIGallery/Controls/PageHeader.xaml (hunk 6 above)
Comment (Jay-o-Way) [reply WinUIGallery/Controls/PageHeader.xaml:131]: "need" seems to be a difficult word here. If it defaults to 16, I don't think it **needs** to. Same discussion for the FontFamily.

File: WinUIGallery/Controls/PageHeader.xaml (hunk 5 above)
Comment (niels9001) [reply WinUIGallery/Controls/PageHeader.xaml:52]: It still looks out of place now. So we'd still need to fix the offset.

Comment (Jay-o-Way) [reply WinUIGallery/Controls/PageHeader.xaml:52]: > this is by design. I'm not changing it :)

File: WinUIGallery/Controls/PageHeader.xaml (hunk 6 above)
Comment (niels9001) [reply WinUIGallery/Controls/PageHeader.xaml:131]: As mentioned above, `FontIcon` defaults to 20. That's what's happening now:

image

When `FontSize` is set to 16 it renders the correct size (with the offset as mentioned earlier):
image

So we'd need to explicitly set it to make sure all icons render the same size.

File: WinUIGallery/Controls/PageHeader.xaml (hunk 5 above)
Comment (Jay-o-Way) [reply WinUIGallery/Controls/PageHeader.xaml:52]: > IIRC Fluent Icons use a 1 pixel thickness for the lines

Try to think of it as **4px** thickness in a **16x16** icon. Then it all adds up. How else would an icon like image be made?

Comment (niels9001) [reply WinUIGallery/Controls/PageHeader.xaml:52]: Right, what I meant was that these icons were probably designed with a 1px grid. Then you end up with 3 pixels above or below in some edge cases, so it's not perfectly aligned.

image

I think with a simple margin tweak we can visually fix that in context with the other icons?

Comment (Jay-o-Way) [reply WinUIGallery/Controls/PageHeader.xaml:52]: I respectfully refuse. If you insist, you're free to do so in a pr of your own :)

File: WinUIGallery/Controls/PageHeader.xaml [hunk 7]
```diff
@@ -48,7 +48,7 @@
                     Visibility="{x:Bind Item.Docs, Converter={StaticResource collectionConverter}}">
                     <DropDownButton.Content>
                         <StackPanel Orientation="Horizontal" Spacing="8">
-                            <FontIcon FontSize="14" Glyph="&#xE8A5;" />
```
Comment (niels9001) [inline WinUIGallery/Controls/PageHeader.xaml:51]: image

This is still a visible regression over the current version as the text label is not vertically aligned wrt the icon and glyph.

FYI, it's very common to add a specific margin to center align text within a control. (e.g. see the default button padding that has a different top/bottom margin to center align the label)

New vs. old:
image

**Review by niels9001 — CHANGES_REQUESTED** [review]

File: WinUIGallery/Controls/PageHeader.xaml (hunk 7 above)
Comment (Jay-o-Way) [reply WinUIGallery/Controls/PageHeader.xaml:51]: Then I would still prefer `VerticalAlignment="Center"`

Comment (Jay-o-Way) [reply WinUIGallery/Controls/PageHeader.xaml:51]: You're comparing the baseline of the icon content with the baseline of text. They don't need to be the same thing.

**Comment by Jay-o-Way** [comment]
@niels9001 can we wrap this up?

**Review by niels9001 — APPROVED** [review]
Thanks

**Pull request merged**